modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/objectbuilder/DefaultObjectBuilder.java

```diff
@@ -120,8 +119,7 @@ public T build(ValueResolvingContext context) throws MuleException {
       final Object resolvedValue = resolveValue(entry.getValue(), context);
 
       entry.getKey().set(object,
-                         context == null || context.resolveCursors() ? resolveCursor(resolvedValue, MessageUtils::decorateInput)
-                             : resolvedValue);
+                         context == null || context.resolveCursors() ? resolveCursor(resolvedValue) : resolvedValue);
```

Comment:
We still need to create a test that would demonstrate that decorating with `UnclosableCursorStream` would be necessary here too. Until then, I'm leaving it like this. But I'm working with @ndinu on this.

Reply:
Already solved on master by #11333. Rebased and adjusted accordingly.